docs/usage/administration/useful_sql_for_admins.md

```diff
@@ -68,43 +99,84 @@ user_directory_search - 316433
 
 ## Show top 20 rooms by new events count in last 1 day:
 ```sql
-SELECT e.room_id, r.name, COUNT(e.event_id) cnt FROM events e
-LEFT JOIN room_stats_state r USING (room_id)
-WHERE e.origin_server_ts >= DATE_PART('epoch', NOW() - INTERVAL '1 day') * 1000 GROUP BY e.room_id, r.name ORDER BY cnt DESC LIMIT 20;
+SELECT
+  e.room_id,
+  r.name,
+  COUNT(e.event_id) cnt 
+FROM
+  events e 
+  LEFT JOIN
+    room_stats_state r USING (room_id) 
+WHERE
+  e.origin_server_ts >= DATE_PART('epoch', NOW() - INTERVAL '1 day') * 1000 
+GROUP BY
+  e.room_id,
+  r.name 
+ORDER BY
+   cnt DESC LIMIT 20;
 ```
 
 ## Show top 20 users on homeserver by sent events (messages) at last month:
```

Comment:
This has the potential to be very slow, I think?

Reply:
Is there a chance for another solution / query? Or should I am add a warning?

Reply:
I think just adding a warning about it would be enough.

Reply:
```bash
synapse=> explain SELECT
synapse->   COUNT(*),
synapse->   sender 
synapse-> FROM
synapse->   events 
synapse-> WHERE
synapse->   (
synapse(>     type = 'm.room.encrypted' 
synapse(>     OR type = 'm.room.message'
synapse(>   )
synapse->   AND origin_server_ts >= DATE_PART('epoch', NOW() - INTERVAL '1 month') * 1000 
synapse-> GROUP BY
synapse->   sender 
synapse-> ORDER BY
synapse->   COUNT(*) DESC LIMIT 20;
                                                                                                              QUERY PLAN                                                                                                               
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Limit  (cost=2077.77..2077.82 rows=20 width=64)
   ->  Sort  (cost=2077.77..2083.53 rows=2303 width=64)
         Sort Key: (count(*)) DESC
         ->  HashAggregate  (cost=1993.46..2016.49 rows=2303 width=64)
               Group Key: sender
               ->  Seq Scan on events  (cost=0.00..1975.47 rows=3598 width=56)
                     Filter: (((type = 'm.room.encrypted'::text) OR (type = 'm.room.message'::text)) AND ((origin_server_ts)::double precision >= (date_part('epoch'::text, (now() - '1 mon'::interval)) * '1000'::double precision)))
(7 rows)
```